src/LAD4_1/lad4_1.cpp

```diff
@@ -1,5 +1,176 @@
+#include "apogeeDetection.h"
 #include <Arduino.h>
+#include <batteryMonitor.h>
+#include "Barometer.h"
+#include "config.h"
+#include "IMU.h"
+#include "GPS.h"
 
-int main(int argc, char** argv) {
-  return 0;
+#define SERIAL_INPUT 0
+
+#if SERIAL_INPUT
+  #define RFSerial Serial
+#else
+  #define RFSerial Serial6
+#endif
+
+#define GPS_Serial Serial8
+
+uint8_t val_index = 0;
+char command[75];
+
+//Stores how often we should be requesting data from each sensor.
+int sensor_checks[numSensors][2];
+
+sensorInfo *sensor;
+
+long startTime;
+String packet;
+
+void sensorReadFunc(int id);
+
+IMU _imu;
+Barometer bmp;
+GPS gps(GPS_Serial);
+ApogeeDetection detector;
+
+double acc_z;
+double altitude;
+
+void setup() {
+  //Setting up Serial Connection
+  Wire.begin();
+  Serial.begin(57600);
+  RFSerial.begin(57600);
+  //Waiting for Serial Connection
+  while(!Serial);
+  while(!RFSerial);
+
+  debug("Setting up Config");
+  config::setup();
+
+  debug("Initializing Sensor Frequencies");
+
+  for (int i = 0; i < numSensors; i++) {
+    sensor_checks[i][0] = sensors[i].clock_freq;
+    sensor_checks[i][1] = 1;
+  }
+
+  debug("Starting SD");
+
+  int res = sd.begin(SdioConfig(FIFO_SDIO));
+  if (!res) {
+    packet = make_packet(101, true);
+    RFSerial.println(packet);
+  }
+
+  debug("Opening File");
+  file.open(file_name, O_RDWR | O_CREAT);
+
+  debug("Writing Dummy Data");
+  // NEED TO DO THIS BEFORE ANY CALLS TO write_to_SD
+  sdBuffer = new Queue();
+
+  std::string start = "beginning writing data";
+  if(!write_to_SD(start, file_name)) { // if unable to write to SD, send error packet
+    packet = make_packet(101, true);
+    RFSerial.println(packet);
+  }
+
+  debug("Initializing Libraries");
+
+  batteryMonitor::init(&Wire, batteryMonitorShuntR, batteryMonitorMaxExpectedCurrent);
+
+
+  _imu.init(&Wire);
+
+  bmp.init(&Wire);
+
+  gps.init();
+
+  // Initializes initial state _x in Kalman filter to the alt and acc of the rocket at setup
+  sensorReadFunc(0);
+  acc_z = farrbconvert.sensorReadings[2];
+  sensorReadFunc(4);
+  altitude = farrbconvert.sensorReadings[0];
+
+  double altVar = 0.5;
+  double accVar = 0.5;
+  detector = ApogeeDetection(20e-3, altVar, accVar, altitude, acc_z);
```

Comment:
If we can't construct this statically, let's modify the apogee detection library to be like the IMU and barometer libraries above w/ a default constructor and a separate init func that we can pass all of this to.
20e-3 should also be a variable defined in config.